nixos/maintainers/scripts/ec2/amazon-image-zfs.nix

```diff
@@ -0,0 +1,13 @@
+{
+  imports = [ ./amazon-image.nix ];
+  ec2.zfs = {
+    enable = true;
+    datasets = {
+      "tank/system/root".mount = "/";
+      "tank/system/var".mount = "/var";
+      "tank/local/nix".mount = "/nix";
+      "tank/user/home".mount = "/home";
+      "boot/boot".mount = "/boot";
```

Comment:
Should there be some form of documentation about the dataset names?

Note that I'm still not familiar with ZFS usage. Maybe it's the only way to do things and totally self-evident.

Reply:
The names I picked are fairly arbitrary. It is very common to have a dataset for each of var, nix, home and /boot. What sort of documentation are you thinking?

Reply:
I don't know. I don't know the expectations with regards to the amazon images generation. I don't know what someone *wanting* to use ZFS would expect to see from a prebuilt image using ZFS.

Reply:
At the very least I'd expect `/nix` to be in its own pool. `/home` and `/var` are fairly standard to pull out as well. I'd say this scheme is expected and useful.